cmd/cmd.go

```diff
@@ -117,26 +119,43 @@ func CreateHandler(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	if len(req.Files) > 0 {
-		fileMap := map[string]string{}
-		for f, digest := range req.Files {
+	var mu sync.Mutex
+	var g errgroup.Group
+	g.SetLimit(max(runtime.GOMAXPROCS(0)-1, 1))
+	// copy files since we'll be modifying the map
+	temp := req.Files
```

Comment:
If we are just going to be overwriting `req.Files` it might make more sense to not set it on the request in the `CreateRequest` function at all, and instead return the slice of files to set on the request in the caller.